### DISCUSSION TITLE
Screenshot on Stellarium

### DISCUSSION THREAD

**Discussion by MayconJD** [Stellarium/stellarium#3932] [Q&A]
Hello ! I'm trying to use Stellarium to simulate satellites passing in front of a telescopic lens. To do this, I need the program to simulate the passage of a satellite at a certain point and take 60 screenshots to simulate the observation time of a telescope. The intention is that I have 60 images and in each of them the satellite passes by and interferes with the observation.
I want Stellarium to take this screenshot for me, however, it is unable to do so. Could anyone help me? Is it possible to do this? I'm using Stellarium's own console to run the code below:

// Set observer location: latitude, longitude, altitude (in meters)
core.setObserverLocation(0.0, 0.0, 100, "", 0);

// Set the current date and time
var now = new Date();
core.setDate(now.toISOString(), "utc");

// Point to a coordinate in the sky (RA, Dec)
core.moveToRaDec(15.0, -30.0);

// Create a satellite (using the ISS as an example)
var satellite = Satellites.createSatelliteFromName("ISS");
satellite.setVisible(true);  // Show the satellite
satellite.setTrajectoryVisible(true);  // Show the satellite's trajectory

// Loop to capture 60 images (shots), 1 per second
for (var i = 1; i <= 60; i++) {
    core.wait(1);  // Wait 1 second between each capture
    core.screenshot("C:\\output\\shot_" + i + ".png");  // Save the screenshot
}

**Comment by alex-w** [comment]
```
// Create a satellite (using the ISS as an example)
var satellite = Satellites.createSatelliteFromName("ISS");
satellite.setVisible(true); // Show the satellite
satellite.setTrajectoryVisible(true); // Show the satellite's trajectory
```

Where is from it?

**Reply by MayconJD** [reply]
The intention was to simulate a satellite passing through the frame I selected.

**Reply by MayconJD** [reply]
Could you help me with a code that does this? As I couldn't find it on the internet, I used AI to try to help me, but to no avail.

**Comment by gzotti** [comment]
> script error: 'Type'  @ line 12: TypeError: Property 'createSatelliteFromName' of object Satellites(0x1c056860a90, "Satellites") is not a function

In which API docs have you seen this function? It may be enough to just search for it.

**Reply by MayconJD** [reply]
Could you point me to an API that does what I need? I searched, but I didn't directly find what I needed

**Reply by gzotti** [reply]
I'd start with
`core.selectObjectByName("ISS")`
or rather
`sat=Satellites.searchByName("ISS")`
as this may return a usable object for further manipulation.

Not sure whether the rest then works.

**Reply by MayconJD** [reply]
Ok. Thanks man
